qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/model.py

```diff
@@ -0,0 +1,701 @@
+        if indices_to_remove:
+            indices_to_remove = sorted(indices_to_remove, reverse=True)
+            for index in indices_to_remove:
+                del line_split[index]
+        return line_split
+
+    def format_element_symbol(self, line_split):
+        # Now convert FE1 to Fe
+        line_split[0] = ''.join([i for i in line_split[0] if not i.isdigit()])
+        if len(line_split[0]) == 2:
+            line_split[0] = ''.join([line_split[0][0].upper(), line_split[0][1].lower()])
+        return line_split
+
+    def read_space_group(self, phase_file_path):
+        space_group = self.find_in_file(phase_file_path, "_symmetry_space_group_name_H-M", '"', '"',
+                                        strip_separator='"')
+        return space_group
+
+    def insert_minus_before_first_digit(self, original_space_group):
+        # https://docs.mantidproject.org/nightly/concepts/PointAndSpaceGroups.html
+        roto_inverted_space_group = original_space_group
+        for character_index, character in enumerate(roto_inverted_space_group):
+            if character.isdigit():
+                split_string = list(roto_inverted_space_group)
+                split_string.insert(character_index, "-")
+                roto_inverted_space_group = "".join(split_string)
+                break  # only apply to first digit
+        return roto_inverted_space_group
+
+    def read_gsas_lst_and_print_wR(self, result_filepath, histogram_data_files):
+        with open(result_filepath, 'rt', encoding='utf-8') as file:
+            result_string = file.read().replace('\n', '')
+            for loop_histogram in histogram_data_files:
+                where_loop_histogram = result_string.rfind(loop_histogram)
+                if where_loop_histogram != -1:
+                    where_loop_histogram_wR = result_string.find('Final refinement wR =', where_loop_histogram)
+                    if where_loop_histogram_wR != -1:
+                        where_loop_histogram_wR_end = result_string.find('%', where_loop_histogram_wR)
+                        logger.notice(loop_histogram)
+                        logger.notice(result_string[where_loop_histogram_wR: where_loop_histogram_wR_end + 1])
+
+    def get_crystal_params_from_instrument(self, instrument):
+        crystal_params = []
+        if not self.number_of_regions:
+            crystal_params = self.find_in_file(instrument, "ICONS", "S", "INS", strip_separator="ICONS\t")
+
+        else:
+            for region_index in range(1, self.number_of_regions+1):
+                loop_crystal_params = self.find_in_file(instrument, f"{region_index} ICONS", "S", "INS",
+                                                        strip_separator="ICONS\t")
+                crystal_params.append(loop_crystal_params)
+        tof_min = []
+        for loop_crystal_param_string in crystal_params:
+            list_crystal_params = loop_crystal_param_string.split("\t")
```

Comment:
If I run a refinement with instrument group=bank 2, focused data=all banks I get an error from this line (`AttributeError: 'NoneType' object has no attribute 'split'`). I think one of the earlier calls to `find_in_file` on L371 or 375 returns None and this gets entered into the crystal_params list when there's a mismatch in the number of banks in the prm and gss files so worth checking if find_in_file returns None here